auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java

```diff
@@ -329,6 +332,33 @@ public void deleteMachineUser(String requestId, String userCrn, String machineUs
         }
     }
 
+    /**
+     * Wraps calls to ListServicePrincipalCloudIdentities with an Account ID.
+     *
+     * @param requestId       the request ID for the request
+     * @param accountId       the account ID
+     * @param environmentCrn  the crn of the environment to list service prinnciples for
+     * @return the list of service principal cloud identities
+     */
+    public List<ServicePrincipalCloudIdentities> listServicePrincipalCloudIdentities(String requestId, String accountId, String environmentCrn) {
+        checkNotNull(requestId);
+        checkNotNull(accountId);
+
+        List<ServicePrincipalCloudIdentities> servicePrincipalCloudIdentities = new ArrayList<>();
+
+        ListServicePrincipalCloudIdentitiesRequest.Builder requestBuilder = ListServicePrincipalCloudIdentitiesRequest.newBuilder()
+                .setAccountId(accountId)
+                .setEnvironmentCrn(environmentCrn);
+
+        UserManagementProto.ListServicePrincipalCloudIdentitiesResponse response;
+        do {
```

Comment:
I recently added paging to the listWorkloadAdministrationGroupsForMember API. I put the paging support in the GrpcUmsClient class instead of the UmsClient class because it was pretty much impossible to unit test in the UmsClient. I suggest moving the paging code to GrpcUmsClient

Reply:
Ahh I see. I've moved it there and added a unit test.